proto/dydxprotocol/vault/query.proto

```diff
@@ -105,12 +112,42 @@ message QueryMegavaultTotalSharesResponse { NumShares total_shares = 1; }
 // QueryMegavaultOwnerSharesRequest is a request type for the
 // MegavaultOwnerShares RPC method.
 message QueryMegavaultOwnerSharesRequest {
-  cosmos.base.query.v1beta1.PageRequest pagination = 3;
+  string address = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
 }
 
 // QueryMegavaultOwnerSharesResponse is a response type for the
 // MegavaultOwnerShares RPC method.
 message QueryMegavaultOwnerSharesResponse {
+  // Owner address.
+  string address = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  // Total number of shares that belong to the owner.
+  NumShares shares = 2 [ (gogoproto.nullable) = false ];
+  // All share unlocks.
+  repeated ShareUnlock share_unlocks = 3 [ (gogoproto.nullable) = false ];
+  // Owner equity in megavault (in quote quantums).
+  bytes equity = 4 [
+    (gogoproto.customtype) =
+        "github.com/dydxprotocol/v4-chain/protocol/dtypes.SerializableInt",
+    (gogoproto.nullable) = false
+  ];
+  // Equity that owner can withdraw in quote quantums (as one cannot
+  // withdraw locked shares).
+  bytes withdrawable_equity = 5 [
+    (gogoproto.customtype) =
+        "github.com/dydxprotocol/v4-chain/protocol/dtypes.SerializableInt",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// QueryMegavaultAllOwnerSharesRequest is a request type for the
+// MegavaultAllOwnerShares RPC method.
+message QueryMegavaultAllOwnerSharesRequest {
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
```

Comment:
What's there to paginate?

Reply:
oh this was the old `MegavaultOwnerSharesRequest` and it lists out all owner shares